sympy/matrices/matrices.py

```diff
@@ -2919,49 +2938,51 @@ def gauss_jordan_solve(self, b, freevar=False):
         """
         from sympy.matrices import Matrix, zeros
 
-        aug = self.hstack(self.copy(), b.copy())
-        row, col = aug[:, :-1].shape
+        aug = self.hstack(self.copy(), B.copy())
+        B_cols = B.shape[1]
```

Comment:
Maybe `B.cols` instead of `B.shape[1]`.

Reply:
Well spotted, thanks. I've changed that.